cnd/src/swap_protocols/rfc003/bitcoin/htlc_events.rs

```diff
@@ -92,12 +92,14 @@ where
         htlc_deployment: &Deployed<htlc_location::Bitcoin, transaction::Bitcoin>,
         start_of_swap: NaiveDateTime,
     ) -> anyhow::Result<Redeemed<transaction::Bitcoin>> {
-        let (transaction, _) =
-            watch_for_spent_outpoint(self, start_of_swap, htlc_deployment.location, vec![vec![
-                1u8,
-            ]])
-            .instrument(tracing::info_span!("htlc_redeemed"))
-            .await?;
+        let (transaction, _) = watch_for_spent_outpoint(
+            self,
+            start_of_swap,
+            htlc_deployment.location,
+            htlc_params.redeem_identity,
+        )
+        .instrument(tracing::info_span!("htlc_redeemed"))
+        .await?;
 
         let secret = extract_secret(&transaction, &htlc_params.secret_hash)
```

Comment:
I should be able to remove this line as well

Reply:
That would be good but I think that is out of scope of this ticket :)

If you have an idea, feel free to record it as an issue!
Thinking about it, I would say the best place for this function would actually be in `blockchain-contracts` because that crate knows, that `SHA256` is used an can therefore search for preimages in the witness stack that, when hashed, result in the given secret hash.

Reply:
I don't think it is causing any problems so I don't think it is worth creating an issue to remove it. I just thought I could simply remote because the watch_outpoint call is already matching the spenders public key to decide whether it is a redeem or refund. The extract_secret check is redundant.